### PR TITLE
Implement msgpack serde for RoleAssignments

### PR DESCRIPTION
**Pull request by Jasopaum**
## Description

We cannot implement protobuf serde because workers can't yet be bufferized.

## Checklist:
* [x] My changes are covered by tests.
* [x] I have run [the pre-commit hooks](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md#setting-up-pre-commit-hook) to format and check my code for style issues.
* [x] I have commented my code following [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

(See the [the contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md) for additional tips.)

### CODE REVIEW

File: syft/execution/role_assignments.py [hunk 1]
```diff
@@ -10,13 +16,20 @@ class RoleAssignments:
...
                 the Protocol to which the RoleAssignments is associated.
         """
-        self.assignments = {role_id: [] for role_id in role_ids}
+        if assignments:
+            self.assignments = assignments
+        elif role_ids:
+            self.assignments = {role_id: [] for role_id in role_ids}
+        else:
+            raise ValueError(
+                "You need to provide role_ids or assignments in RoleAssignments' constructor"
```
Comment (tudorcebere) [inline syft/execution/role_assignments.py:31]: I think this is a typo, should be RoleAssignament's.

File: syft/execution/role_assignments.py [hunk 2]
```diff
@@ -10,13 +16,20 @@ class RoleAssignments:
     which other parties are participating and communicate with them if needed.
     """
 
-    def __init__(self, role_ids):
+    def __init__(self, role_ids: list = [], assignments: dict = {}):
```
Comment (tudorcebere) [inline syft/execution/role_assignments.py:19]: I think there might be a hidden bug when settings obj refs as default arguments (might not occur here, but I would personally avoid it just for future usage).

```
class Test:
    def __init__(self, test={}):
        self.test = test       

    def add_elem(self, k, v):
        self.test[k] = v

    def print(self):
        print(self.test)

t1 = Test()
t1.add_elem("1", 1)
t1.print()
# {"1": 1}

t2 = Test()
t2.add_elem("2", 2)
t2.print()
# {"1": 1, "2": 2}         
```

File: syft/execution/role_assignments.py (hunk 1 above)
Comment (Jasopaum) [reply syft/execution/role_assignments.py:28]: The class name is plural, so there is no typo, I think (English is not my native language so I may be wrong though ^^)

File: syft/execution/role_assignments.py (hunk 2 above)
Comment (Jasopaum) [reply syft/execution/role_assignments.py:16]: Oh yep, I already had this kind of bug! Nice catch 👍

File: syft/execution/role_assignments.py (hunk 1 above)
Comment (karlhigley) [reply syft/execution/role_assignments.py:28]: Bizarrely, this is the correct punctuation in English. It's a shit language and I make no excuses.

**Pull request merged**